Kitodo/src/main/java/org/kitodo/production/helper/tasks/HierarchyMigrationTask.java

```diff
@@ -0,0 +1,358 @@
+     * format using XSLT.
+     *
+     * @param process
+     *            process to migrate
+     */
+    private static void migrateMetadataFiles(Process process) throws IOException {
+        URI metadataFilePath = fileService.getMetadataFilePath(process);
+        dataEditorService.readData(metadataFilePath);
+        URI anchorFilePath = fileService.createAnchorFile(metadataFilePath);
+        dataEditorService.readData(anchorFilePath);
+    }
+
+    /**
+     * Reads the parent record identifier from the anchor file.
+     */
+    private static Optional<String> getParentRecordId(Process process) throws IOException {
+        URI metadataFilePath = fileService.getMetadataFilePath(process);
+        URI anchorFilePath = fileService.createAnchorFile(metadataFilePath);
+        Workpiece anchorWorkpiece = metsService.loadWorkpiece(anchorFilePath);
+        Optional<String> parentRecordId = anchorWorkpiece.getRootElement().getMetadata().parallelStream()
+                .filter(metadata -> metadata.getKey().equals("CatalogIDDigital"))
+                .filter(MetadataEntry.class::isInstance).map(MetadataEntry.class::cast).map(MetadataEntry::getValue)
+                .findFirst();
+        return parentRecordId;
+    }
+
+    /**
+     * Creates a new parent process. The process is created in the database, the
+     * process folder is created on the file system and the METS file is
+     * written. The METS file of the child process is migrated and a link
+     * between the processes is written in the database.
+     *
+     * @param childProcess
+     *            process to migrate
+     * @return a data object with the ID of the created parent process and the
+     *         current number of the child process
+     */
+    private List<Integer> createParentProcess(Process childProcess)
+            throws ProcessGenerationException, IOException, DataException {
+
+        processGenerator.generateProcess(childProcess.getTemplate().getId(), childProcess.getProject().getId());
+        Process parentProcess = processGenerator.getGeneratedProcess();
+        processService.save(parentProcess);
+        fileService.createProcessLocation(parentProcess);
+        createParentMetsFile(childProcess);
+        ArrayList<Integer> parentData = new ArrayList<>();
+        parentData.add(parentProcess.getId());
+        URI metadataFilePath = fileService.getMetadataFilePath(childProcess);
+        parentData.add(convertChildMetsFile(metadataFilePath));
+        linkInDatabase(parentProcess, childProcess);
+        return parentData;
+    }
+
+    /**
+     * Links parent process and child process in the database. The processes are
+     * saved.
+     *
+     * @param parentProcess
+     *            parent process to link
+     * @param childProcess
+     *            child process to link
+     */
+    private static void linkInDatabase(Process parentProcess, Process childProcess) throws DataException {
```

Comment:
Its not only in Database but in elasticsearch and Object level.
```suggestion
    private static void linkParentProcessWithChildProcess(Process parentProcess, Process childProcess) throws DataException {
```

Reply:
What I wanted to say is that the link is created at the database level (object, Elasticsearch, is synonymous) in contrast to METS pointer link. _Link parent process with child process_ is not clear, which of the two links is created here.